types/ember__routing/-private/transition.d.ts

```diff
@@ -1,12 +1,26 @@
 import RouteInfo from './route-info';
 
-export default interface Transition {
+export default interface Transition<T = any> extends Partial<Promise<T>> {
```

Comment:
I took this from

https://github.com/tildeio/router.js/blob/bb16c56396ef68416fac80fe7e4fb493ce2bed42/lib/router/transition.ts#L42

Which also exports a `type PublicTransition = Transition<any>;`

https://github.com/tildeio/router.js/blob/bb16c56396ef68416fac80fe7e4fb493ce2bed42/lib/router/transition.ts#L20

I left the type arguments in, as they may be useful in the future?